pkg/acquisition/modules/wineventlog/wineventlog_windows.go

```diff
@@ -0,0 +1,356 @@
+		fragment, err := winlog.RenderFragment(event, wevtapi.EvtRenderEventXml)
+		if err != nil {
+			w.logger.Errorf("Failed to render event with RenderFragment, skipping: %v", err)
+			continue
+		}
+		w.logger.Tracef("Rendered event: %s", fragment)
+		renderedEvents = append(renderedEvents, fragment)
+	}
+	return renderedEvents, err
+}
+
+func (w *WinEventLogSource) buildXpathQuery() (string, error) {
+	var query string
+	queryComponents := [][]string{}
+	if w.config.EventIDs != nil {
+		eventIds := []string{}
+		for _, id := range w.config.EventIDs {
+			eventIds = append(eventIds, fmt.Sprintf("EventID=%d", id))
+		}
+		queryComponents = append(queryComponents, eventIds)
+	}
+	if w.config.EventLevel != "" {
+		levels, err := logLevelToInt(w.config.EventLevel)
+		logLevels := []string{}
+		if err != nil {
+			return "", err
+		}
+		for _, level := range levels {
+			logLevels = append(logLevels, fmt.Sprintf("Level=%s", level))
+		}
+		queryComponents = append(queryComponents, logLevels)
+	}
+	if len(queryComponents) > 0 {
+		andList := []string{}
+		for _, component := range queryComponents {
+			andList = append(andList, fmt.Sprintf("(%s)", strings.Join(component, " or ")))
+		}
+		query = fmt.Sprintf("*[System[%s]]", strings.Join(andList, " and "))
+	} else {
+		query = "*"
+	}
+	queryList := QueryList{Select: Select{Path: w.config.EventChannel, Query: query}}
+	xpathQuery, err := xml.Marshal(queryList)
+	if err != nil {
+		w.logger.Errorf("Marshal failed: %v", err)
+		return "", err
+	}
+	w.logger.Debugf("xpathQuery: %s", xpathQuery)
+	return string(xpathQuery), nil
+}
+
+func (w *WinEventLogSource) getEvents(out chan types.Event, t *tomb.Tomb) error {
+	subscription, err := winlog.Subscribe(w.evtConfig)
+	if err != nil {
+		w.logger.Errorf("Failed to subscribe to event log: %s", err)
+		return err
+	}
+	defer winlog.Close(subscription)
+	publisherCache := make(map[string]windows.Handle)
+	defer func() {
+		for _, h := range publisherCache {
+			winlog.Close(h)
+		}
+	}()
+	for {
+		select {
+		case <-t.Dying():
+			w.logger.Infof("wineventlog is dying")
+			return nil
+		default:
+			status, err := windows.WaitForSingleObject(w.evtConfig.SignalEvent, 1000)
+			if err != nil {
+				w.logger.Errorf("WaitForSingleObject failed: %s", err)
+				return err
+			}
+			if status == syscall.WAIT_OBJECT_0 {
+				renderedEvents, err := w.getXMLEvents(w.evtConfig, publisherCache, subscription, 500)
+				if err == windows.ERROR_NO_MORE_ITEMS {
+					windows.ResetEvent(w.evtConfig.SignalEvent)
+				} else if err != nil {
+					w.logger.Errorf("getXMLEvents failed: %v", err)
+					continue
+				}
+				for _, event := range renderedEvents {
+					linesRead.With(prometheus.Labels{"source": w.name}).Inc()
+					l := types.Line{}
+					l.Raw = event
+					l.Module = w.GetName()
+					l.Labels = w.config.Labels
+					l.Time = time.Now()
+					l.Src = w.name
+					l.Process = true
+					if !w.config.UseTimeMachine {
```

Comment:
just a slight suggestion : we could have a helper at the `config` obj level to deal with the LIVE / TIMEMACHINE FLAG in the object ?